doc/schema.md

```diff
@@ -0,0 +1,190 @@
+As an example, here is the schema for the `address` field of a
+[Patient](https://hl7.org/fhir/patient-definitions.html#Patient.address)
+resource:
+
+```
+optional group address (LIST) {
+  repeated group array {
+    optional binary use (STRING);
+    optional binary type (STRING);
+    optional binary text (STRING);
+    optional group line (LIST) {
+      repeated binary array (STRING);
+    }
+    optional binary city (STRING);
+    optional binary district (STRING);
+    optional binary state (STRING);
+    optional binary postalCode (STRING);
+    optional binary country (STRING);
+    optional group period {
+      optional binary start (STRING);
+      optional binary end (STRING);
+    }
+  }
+}
+```
+
+### Choice types
+
+A FHIR "choice type", i.e., fields ending with `[x]` which can take multiple
+types, are modeled as a record. The fields of the record are
+named after the possible types. For example,
+[`Patient.deceased[x]`](https://hl7.org/fhir/patient-definitions.html#Patient.deceased_x_)
+can be a `boolean` or a `dateTime`; hence it is modeled with the following
+Parquet schema:
+
+```
+optional group deceased {
+  optional boolean boolean;
+  optional binary dateTime (STRING);
+}
+```
+
+### References
+
+FHIR [references](https://hl7.org/fhir/references.html#Reference) are also
+records but because they frequently participate in `JOIN` queries between
+different resource tables, they have some extra special fields. These fields
+represent each resource type that a reference can refer to and make it easier to
+write `JOIN` queries. For example, the
+[`Patient.generalPractitioner`](https://hl7.org/fhir/patient-definitions.html#Patient.generalPractitioner)
+can be a reference to an `Organization` or `Practitioner` or `PractitionerRole`.
+Therefor, it is mapped to the following Parquet schema (only special fields are
+shown; note there might be multiple `generalPractitioner`, hence the `LIST`):
+
+```
+optional group generalPractitioner (LIST) {
+  repeated group array {
+    optional binary organizationId (STRING);
+    optional binary practitionerId (STRING);
+    optional binary practitionerRoleId (STRING);
+    ... [rest of the usual fields]
+  }
+}
+```
+
+### Recursion
+
+When mapping FHIR types to Parquet schema, we sometime need to break recursive
+structures. For example, a FHIR
+[references](https://hl7.org/fhir/references.html#Reference) has an `identifier`
+field which has an
+[`assigner`](https://hl7.org/fhir/datatypes-definitions.html#Identifier.assigner)
+field which is a reference itself. Therefor, there is a
+[`recursiveDepth`](https://github.com/google/fhir-data-pipes/blob/6b8cc412331de948eb1fa16d6a84b31a0cea9fc8/pipelines/batch/src/main/java/com/google/fhir/analytics/BasePipelineOptions.java#L64)
+configuration parameter that controls how many times a recursive type should
+be traversed in the same branch.
+
+### Extensions
```

Comment:
When I was working on the initial version, I thought about this but decided against it as I thought it might be too much details. But now per your comment, I added a brief note about it. Please let me know if it is not very clear. BTW, can you please also remind me of a good US-Core example that demonstrates this, i.e., two extensions for the same type but in different `StructureDefinition` files?

Reply:
I had a quick check and couldn't get the example. Will check later and update you on the same.

Reply:
Okay thanks; then let's add the example later.